File: dotnet/Allmark/Render/BlockQuoteRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class BlockQuoteRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "block_quote",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderTag.Execute(node, state, "blockquote");
    }
}

File: zig/src/render/blockQuoteRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderTagFn = @import("renderTag.zig").renderTag;

pub fn render(node: *const MarkdownNode, state: *RendererState, first: ?bool, last: ?bool, decode: ?bool) void {
    _ = first;
    _ = last;
    _ = decode;
    renderTagFn(node, state, "blockquote", true);
}

pub const blockQuoteRenderer = Renderer{
    .name = "block_quote",
    .render = render,
};
